pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -975,13 +975,40 @@ class Timedelta(_Timedelta):
         __rdiv__ = __rtruediv__
 
     def __floordiv__(self, other):
+        # numpy does not implement floordiv for timedelta64 dtype, so we cannot
+        # just defer
+        if hasattr(other, '_typ'):
+            # Series, DataFrame, ...
+            return NotImplemented
+
         if hasattr(other, 'dtype'):
-            # work with i8
-            other = other.astype('m8[ns]').astype('i8')
-            return self.value // other
+            if other.dtype.kind == 'm':
+                # also timedelta-like
+                # We need to watch out for np.timedelta64('NaT')
+                if np.ndim(other) == 0:
+                    if np.isnat(other):
```

Comment:
pls try to be DRY
don’t use np.ndim ; we don’t use it anywhere else
don’t use np.isnat

Reply:
> don’t use np.isnat

This is specifically trying to catch `np.timedelta64('NaT')` and because of np.NaT rules we can't use `is timedelta64('NaT')` and we can't use `== timedelta64('NaT')`